tcp/tcp.go

```diff
@@ -1 +1,223 @@
-package tcp
\ No newline at end of file
+package tcp
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/renproject/aw/handshake"
+	"github.com/renproject/aw/protocol"
+	"github.com/sirupsen/logrus"
+)
+
+type Client struct {
+	logger logrus.FieldLogger
+	pool   ConnPool
+}
+
+func NewClient(logger logrus.FieldLogger, pool ConnPool) *Client {
+	return &Client{
+		logger: logger,
+		pool:   pool,
+	}
+}
+
+func (client *Client) Run(ctx context.Context, messages protocol.MessageReceiver) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case messageOtw := <-messages:
+			go client.handleMessageOnTheWire(messageOtw)
```

Comment:
Should we spawn a goroutine for every message? This feels like it would have massive overhead for a large number of messages all sent at once. (Perhaps we should setup a benchmark for this scenario.)